app/src/main/java/org/simple/clinic/drugs/EditMedicinesEffectHandler.kt

```diff
@@ -42,16 +43,20 @@ class EditMedicinesEffectHandler @AssistedInject constructor(
 
       val protocolDrugsStream = effects
           .observeOn(io)
-          .flatMap { userSession.requireLoggedInUser() }
-          .switchMap { facilityRepository.currentFacility(it) }
-          .switchMap { protocolRepository.drugsForProtocolOrDefault(it.protocolUuid) }
+          .map { currentProtocolUuid() }
+          .switchMap { protocolRepository.drugsForProtocolOrDefault(it) }
 
       val prescribedDrugsStream = effects
-          .flatMap { prescriptionRepository.newestPrescriptionsForPatient(it.patientUuid).subscribeOn(io) }
+          .switchMap { prescriptionRepository.newestPrescriptionsForPatient(it.patientUuid).subscribeOn(io) }
 
       Observables
           .combineLatest(protocolDrugsStream, prescribedDrugsStream)
           .map { DrugsListFetched(it.first, it.second) }
     }
   }
+
+  private fun currentProtocolUuid(): UUID? {
```

Comment:
This need not be null since we'll generally have a protocol UUID. Otherwise, if we expect this to actually be nullable, we can return an `Optional` instead and filter the stream since Rx will crash if we return a `null`.

Reply:
This can be `null` because the `currentFacility` API returns a nullable. 
A better way to do this is to not propagate nullable facility post login because it's not going to happen. I can add null-safety in this method instead.